### PR TITLE
Modifications for Openshift

### PR DESCRIPTION
**Pull request by tobiasge**
As requested in PR #46 I have now placed the default configuration in the container.
I also updated the README.md to indicate in wich directory the configuration files must be mapped.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -39,6 +39,11 @@ COPY docker/docker-entrypoint.sh docker-entrypoint.sh
 COPY startup_scripts/ /opt/netbox/startup_scripts/
 COPY initializers/ /opt/netbox/initializers/
 
+# Place default configuration in the container, so that it can be used
+# without docker-compose or older versions of docker-compose.yml
+RUN mkdir /etc/netbox
```
Comment (cimnine) [inline Dockerfile]: this is not necessary according to the [Dockerfile reference][dockerfile]:

> * If `<dest>` doesn’t exist, it is created along with all missing directories in its path.

[dockerfile]: https://docs.docker.com/engine/reference/builder/#copy

File: Dockerfile [hunk 2]
```diff
@@ -39,6 +39,11 @@ COPY docker/docker-entrypoint.sh docker-entrypoint.sh
 COPY startup_scripts/ /opt/netbox/startup_scripts/
 COPY initializers/ /opt/netbox/initializers/
 
+# Place default configuration in the container, so that it can be used
+# without docker-compose or older versions of docker-compose.yml
```
Comment (cimnine) [inline Dockerfile]: Please remove the comments. IMO they don't tell us anything that we don't already know. (Any of the files can be replaces using a docker-compose mounting, so why bother writing this down?)

Comment (cimnine) [reply Dockerfile]: (Then you can get rid of the extra blank link as well)

File: README.md [hunk 3]
```diff
@@ -151,7 +151,7 @@ You should therefore adjust the configuration for production setups, at least th
 
 You may run this image in a cluster such as Docker Swarm, Kubernetes or OpenShift, but this is advanced level.
 
-In this case, we encourage you to statically configure Netbox by starting from [Netbox's example config file][default-config], and mounting it into your container using the mechanism provided by your container platform (i.e. [Docker Swarm configs][swarm-config], [Kubernetes ConfigMap][k8s-config], [OpenShift ConfigMaps][openshift-config]).
+In this case, we encourage you to statically configure Netbox by starting from [Netbox's example config file][default-config], and mounting it into your container in the directory /etc/netbox/ using the mechanism provided by your container platform (i.e. [Docker Swarm configs][swarm-config], [Kubernetes ConfigMap][k8s-config], [OpenShift ConfigMaps][openshift-config]).
```
Comment (cimnine) [inline README.md]: Please add a pari of backticks (` `` `) around `/etc/netbox/`

**Comment by tobiasge** [comment]
I made the requested changes.

**Comment by cimnine** [comment]
Thanks again for your contribution. The build of the new Docker images usually takes about 30 minutes.

**Comment by cimnine** [comment]
One more thing: Can I now drop this part from the README? It works on your OpenShift, right?

> **A Note On OpenShift**
> 
> OpenShift usually is configured with specific restrictions regarding root users. Special care has to be taken when building images for OpenShift. The Docker Image that may be built using this project (and which is available on Docker Hub) might not yet run without further customization on OpenShift. If you have this running on OpenShift, it would be nice if you could open a PR with the changes you needed to make. Or if you didn't do any changes and it just worked, that you could confirm this so that we can remove this notice.

**Comment by tobiasge** [comment]
Yes, it works on Openshift now (We're using it on Openshift Enterprise 3.7, including the LDAP changes for which I will create a new PR)

**Pull request merged**